pkg/kubernetes/provider_kubeconfig.go

```diff
@@ -118,14 +123,19 @@ func (p *kubeConfigClusterProvider) GetDefaultTarget() string {
 	return p.defaultContext
 }
 
-func (p *kubeConfigClusterProvider) WatchTargets(onKubeConfigChanged func() error) {
-	m := p.managers[p.defaultContext]
-	m.WatchKubeConfig(onKubeConfigChanged)
-	m.WatchClusterState(DefaultClusterStatePollInterval, DefaultClusterStateDebounceWindow, onKubeConfigChanged)
+func (p *kubeConfigClusterProvider) WatchTargets(reload McpReload) {
+	reloadWithCacheInvalidate := func() error {
+		// Invalidate all cached managers to force reloading on next access
+		for contextName := range p.managers {
```

Comment:
do we need a nil check here if case the manager is nil?

Reply:
👍 good catch